File: frontend/src/lib/providerIa.ts
// Qual IA atende o clique: a assinatura do Claude (`claude -p`) ou a do
// Antigravity (`agy -p`). O backend aceita o mesmo par em todas as etapas.

export type ProviderIA = 'claude' | 'gemini';

/**
 * O provider que está gerando agora, ou `null` quando nada está em voo.
 *
 * As mutations de IA guardam o provider no `variables`, e é dele que a tela
 * sabe QUAL botão deve girar — sem isso os dois giram no mesmo clique.
 */
export function providerEmVoo(mutation: {
  isPending: boolean;
  variables?: ProviderIA | void;
}): ProviderIA | null {
  return mutation.isPending ? ((mutation.variables as ProviderIA) ?? 'claude') : null;
}

/**
 * Qual assinatura atendeu, a partir do nome do modelo gravado.
 *
 * É assim que uma geração antiga ganha selo sem coluna nova em cada tabela: o
 * nome do modelo já viaja na telemetria e na avaliação do bruto.
 */
export function providerDoModelo(modelo: string | null | undefined): ProviderIA | null {
  if (!modelo) return null;
  return modelo.toLowerCase().includes('gemini') ? 'gemini' : 'claude';
}

File: backend/app/domain/compartilhado/provider_ia.py
"""Qual IA atende uma geração: a assinatura do Claude ou a do Antigravity.

Módulo próprio, e não uma constante em `services/claude_ia`, porque os routers
precisam do TIPO no topo do arquivo (é anotação de query param) enquanto importam
o serviço tarde, dentro da função, para não fechar ciclo de import. Aqui não há
dependência nenhuma, então qualquer camada pode importar sem risco.
"""

from typing import Literal

ProviderIA = Literal["claude", "gemini"]


def provider_do_modelo(model: str | None) -> ProviderIA | None:
    """Qual assinatura atendeu, a partir do nome do modelo gravado.

    É assim que a tela marca uma geração antiga sem precisar de coluna nova em
    cada tabela: o nome do modelo já viaja na telemetria e na avaliação do bruto.
    """
    if not model:
        return None
    return "gemini" if "gemini" in model.lower() else "claude"
